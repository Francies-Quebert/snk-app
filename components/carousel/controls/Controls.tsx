import SimpleLineIcons from "@expo/vector-icons/SimpleLineIcons";
import { Pressable, StyleProp, TextStyle, ViewStyle } from "react-native";
import { styles } from "../styles";
import { theme } from "@/styles/theme";

interface ControlButtonProps {
  onPress: () => void;
  iconName: string;
  style?: StyleProp<ViewStyle>;
  iconStyle?: StyleProp<TextStyle>;
  testID?: string;
}

export const ControlButton = ({ onPress, iconName, style, iconStyle, testID }: ControlButtonProps) => {
  return (
    <Pressable style={[styles.controlButton, style]} onPress={onPress} {...(testID && { testID })}>
      <SimpleLineIcons name={iconName as any} size={32} style={[styles.controlIcon, iconStyle]} color={theme.colors["dark-primary"]} />
    </Pressable>
  );
};
