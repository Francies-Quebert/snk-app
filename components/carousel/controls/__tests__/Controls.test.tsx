import React from "react";
import { render } from "@testing-library/react-native";
import { ControlButton } from "../Controls";

// Mock the SimpleLineIcons component
jest.mock("@expo/vector-icons/SimpleLineIcons", () => "SimpleLineIcons");

describe("ControlButton", () => {
  it("renders correctly", () => {
    const mockOnPress = jest.fn();
    const { toJSON } = render(<ControlButton onPress={mockOnPress} iconName="arrow-right" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders with custom styles", () => {
    const mockOnPress = jest.fn();
    const { toJSON } = render(<ControlButton onPress={mockOnPress} iconName="arrow-left" style={{ backgroundColor: "red" }} iconStyle={{ color: "blue" }} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
