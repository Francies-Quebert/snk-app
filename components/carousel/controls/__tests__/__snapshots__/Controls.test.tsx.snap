// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 20,
        "padding": 5,
        "position": "absolute",
        "top": "50%",
        "transform": [
          {
            "translateY": -16,
          },
        ],
      },
      undefined,
    ]
  }
>
  <SimpleLineIcons
    color="#1E1E1E"
    name="arrow-right"
    size={32}
    style={
      [
        {
          "color": "#000",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`ControlButton renders with custom styles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 20,
        "padding": 5,
        "position": "absolute",
        "top": "50%",
        "transform": [
          {
            "translateY": -16,
          },
        ],
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <SimpleLineIcons
    color="#1E1E1E"
    name="arrow-left"
    size={32}
    style={
      [
        {
          "color": "#000",
        },
        {
          "color": "blue",
        },
      ]
    }
  />
</View>
`;
